runner_service/controllers/hosts.py

```diff
@@ -159,16 +176,19 @@ def post(self, host_name, group_name):
         ```
         """ # noqa
 
-        valid_parms = ['others']
+        valid_parms = ['others', 'port']
         group_list = []
         group_list.append(group_name)
+        ssh_port = None
 
         args = request.args.to_dict()
+
         if args:
             logger.debug("additional args received")
             if all(p in valid_parms for p in args.keys()):
                 if 'others' in args:
                     group_list.extend(args['others'].split(','))
+                ssh_port = args.get('port', None)
             else:
```

Comment:
Sorry - I didn't notice this before, but can you validate the port before we pass it on to use it in the inventory - i.e. must be numeric

Reply:
Done.